src/pybind/mgr/dashboard/services/cephfs.py

```diff
@@ -55,6 +55,13 @@ def opendir(self, dirpath):
         try:
             d = self.cfs.opendir(dirpath)
             yield d
+        # TODO: Expected cephfs.PremissionError # pylint: disable=fixme
+        except cephfs.OSError as err:
+            if isinstance(dirpath, bytes):
+                dirpath = dirpath.decode()
+            dirpath = dirpath.replace("/.snap", "")
+            err.args = (*err.args, dirpath)
+            raise err
```

Comment:
While Python is extremely liberal regarding what you can do with an object, I'd suggest that we treat object as immutable entities (I'm not advocating here for functional programming, I'm just saying that an object should not contain properties/members that are not declared in their originating class).

In this very case, while Python `OSError` allows you to define the `filename`, `cephfs.OSError` does not, hence we're extending implicitly the interface of the exception at object level (not class level), and a golden rule in Python (and many other languages) is "explicit over implicit". If we want to have a `OSError` class let's have a new exception inspired in `DashboardExcetpion` and Python `OSError`:
```py
# dashboard.exceptions file
class DashboardCephfsError(DashboardException):
  def __init__(self, errno, strerror, filename=None, **kwargs):
    super().__init__(code=errno, component='cephfs', http_status_code=400, msg=strerror)
    self.filename = filename
```